### ISSUE TITLE
RandomGenerator.Next(decimal, decimal) is broken for certain cultures

### ISSUE THREAD

**Issue by anton-gogolev** [nbuilder/nbuilder#50]
Currently, `RandomGenerator.Next(decimal, decimal)` does the following:

```
return (decimal)Convert.ToDecimal(string.Format("{0}.{1}", integer, fraction));
```

This fails for cultures where value of `NumberFormatInfo.NumberDecimalSeparator` is anything other than `.`.

One option would be replacing the `.` with the value of `NumberFormatInfo.NumberDecimalSeparator`, but that would still incur an awkward "integer to string to decimal" roundtrip.

Here's what I think is a better way:

```
public virtual decimal Next(decimal min, decimal max)
{
    if (min < int.MinValue)
        min = (decimal)int.MinValue;

    if (max > int.MaxValue)
        max = (decimal)int.MaxValue;

    var scale = (decimal)Next(0d, 1d);
    var increment = scale * (max - min);

    return min + increment;
}
```

**Comment by crmckenzie** [comment]
This is a duplicate of [this issue](https://github.com/nbuilder/nbuilder/issues/19). The fix is on the develop branch waiting on me to get a free afternoon to finalize the release and push it out. Anton, if you would like to verify that we've fixed the issue correctly, please take a look at the commits. I'll leave this issue open pending your feedback or until I release--whichever comes first.

**Comment by anton-gogolev** [comment]
If you're referring to 65306b3ed5691074f314dba12ef985c204044ab0, than yes, this indeed fixes the issue.  Thanks!
